chapter8.adoc

```diff
@@ -1,37 +1,124 @@
-`MSDCFG.sdedbgalw` is 1, then the external debug mode may be entered but is
-restricted by `M-mode` software to prevent privileged CSR accesses and memory
-accesses by instructions executed in external debug mode cannot use `M` privilege.
-When `medbgen` is 0 and `MSDCFG.sdedbgalw` is also 0, the M-mode root domain
-security manager must not configure triggers with `action=1`. Triggers for the
-supervisor domain are expected to be controlled by the root domain security
-manager to prevent any leakage of information.
+=== `Smsdetrc`: External Trace allowed control for Supervisor Domain
+
+When M-mode external trace is enabled, all supervisor domains activity may also
+be traced by an external trace tool irrespective of the configuration held in
+`msdcfg.sdetralw`.
+
+When M-mode external trace is disabled, whether execution at privilege modes
+less than `M-mode` may be traced by an external trace tool depends on the
+configuration held in `msdcfg.sdetrcalw`, as described below:
+
+When `msdcfg.sdetrcalw` = 0, external trace is disallowed for the SD. All trace
+output from the trace encoder module are disabled while the SD is active on a
+per-hart basis.
+
+When `msdcfg.sdetrcalw` = 1 then external trace of privilege modes less than
+`M-mode` shall be allowed for the SD on a per hart basis.
+
+To enforce the above controls specified by this extension, the following
+functional and security requirements must be met by the external secure debug
+system cite:[ExtDbgSec] to meet the security objectives of supervisor domain
+isolation:
+
+. The enable control for external trace driven by the external trace module is
+  expected to be established by the platform root-of-trust following RISC-V
+  Security Model recommendations SR_GEN_007 and SR_GEN_012. When the control is
+  in cleared state, the hart should not be able to generate trace.
+. Per the Efficient trace specification cite:[ETrc], the side-band `halted`
+  signal being asserted, stops subsequent tracing from the hart. On this signal
+  being deasserted, the encoder can start tracing again. Implementation of this
+  `halted` side-band signal is required to support external tracing with
+  supervisor domains. The state of the `halted` side-band signal shall be gated
+  by the external tracing enable control for M-mode.
+. When external tracing is disabled, tracing should be stopped (via asserting
+  `halted`) on:
+.. entry to Debug mode
+.. entry to `M-mode` privilege
+.. entry to privilege less than `M-mode` with `msdcfg.sdetrcalw` = 0.
+. When extracing tracing is disabled, tracing should be started (via deasserting
+  `halted`) on:
+.. Entry to privilege less than `M-mode` with `msdcfg.sdetrcalw` = 0
+
+When external tracing is enabled (and authorized), there are no restrictions and
+all privilege modes (inclusive of `M-mode` and any supervisor domains managed by
+the RDSM) are traceable. In this case, tracing shall be enabled (via asserting
```

Comment:
all privilege modes (inclusive of `M-mode`  -> this is not right. This section only talks about SD and should not make a statement that enabling tracing for SD enables tracing of M-mode.

Reply:
sorry I meant when M-mode external tracing is enabled - anyway - this comment is not needed any more I think.